wcfsetup/install/files/lib/system/session/SessionHandler.class.php

```diff
@@ -147,18 +125,40 @@ class SessionHandler extends SingletonFactory {
 	 */
 	protected $usersOnlyPermissions = [];
 	
+	private const ACP_SESSION_LIFETIME = 7200;
+	private const GUEST_SESSION_LIFETIME = 7200;
+	private const USER_SESSION_LIFETIME = 86400 * 14;
+	
 	/**
 	 * Provides access to session data.
 	 * 
 	 * @param	string		$key
 	 * @return	mixed
 	 */
 	public function __get($key) {
-		if (isset($this->environment[$key])) {
-			return $this->environment[$key];
+		if ($key === 'sessionID') {
+			return $this->sessionID;
 		}
+		else if ($key === 'userID') {
+			return $this->user->userID;
+		}
+		// TODO: pageID, pageObjectID, parentPageID, parentPageObjectID
 		
-		return $this->session->{$key};
+		/** @deprecated	5.4 - These values can be retrieved more efficiently by directly using the methods in e.g. UserUtil. */
+		$environment = [
+			'ipAddress' => UserUtil::getIpAddress(),
+			'userAgent' => UserUtil::getUserAgent(),
+			'requestURI' => UserUtil::getRequestURI(),
+			'requestMethod' => !empty($_SERVER['REQUEST_METHOD']) ? substr($_SERVER['REQUEST_METHOD'], 0, 7) : '',
+			'spiderID' => $this->getSpiderID(UserUtil::getUserAgent()),
+			'lastActivityTime' => TIME_NOW,
+		];
+		
+		if (array_key_exists($key, $environment)) {
+			return $environment[$key];
+		}
+		
+		throw new \BadMethodCallException("Unknown key '".$key."'");
```

Comment:
```suggestion
		return $environment[$key] ?? null;
```

Your change modified the behavior of the method which would previously returned `null` for unknown keys.